### PR TITLE
Hardening scanner

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

File: sonar-batch/src/main/java/org/sonar/batch/scan/filesystem/FileIndexer.java [hunk 1]
```diff
@@ -183,6 +187,51 @@ private boolean accept(InputFile inputFile) {
...
+    IndexFileVisitor(InputFileBuilder inputFileBuilder, DefaultModuleFileSystem fileSystem, Progress status, InputFile.Type type) {
+      this.inputFileBuilder = inputFileBuilder;
+      this.fileSystem = fileSystem;
+      this.status = status;
+      this.type = type;
+    }
+
+    @Override
+    public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
+      if (Files.isHidden(dir) || dir.getFileName().toString().startsWith(".")) {
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/scan/filesystem/FileIndexer.java]: MINOR Use charAt(int) instead [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS3027)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Comment by henryju** [comment]
Few comments but LGTM

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.

File: sonar-batch/src/main/java/org/sonar/batch/scan/filesystem/FileIndexer.java [hunk 2]
```diff
@@ -183,6 +188,56 @@ private boolean accept(InputFile inputFile) {
...
+
+    IndexFileVisitor(InputFileBuilder inputFileBuilder, DefaultModuleFileSystem fileSystem, Progress status, InputFile.Type type) {
+      this.inputFileBuilder = inputFileBuilder;
+      this.fileSystem = fileSystem;
+      this.status = status;
+      this.type = type;
+    }
+
+    @Override
+    public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/scan/filesystem/FileIndexer.java]: MINOR Add a length check and use "charAt(0)" instead. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS3027)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Pull request merged**